pkg/local_object_storage/metabase/put.go

```diff
@@ -349,48 +348,43 @@ func putListIndexItem(tx *bbolt.Tx, item namedBucketItem) error {
 
 // encodeList decodes list of bytes into a single blog for list bucket indexes.
 func encodeList(lst [][]byte) ([]byte, error) {
-	buf := bytes.NewBuffer(nil)
-	encoder := gob.NewEncoder(buf)
-
-	// consider using protobuf encoding instead of glob
-	if err := encoder.Encode(lst); err != nil {
-		return nil, err
+	w := io.NewBufBinWriter()
+	w.WriteVarUint(uint64(len(lst)))
+	for i := range lst {
+		w.WriteVarBytes(lst[i])
 	}
-
-	return buf.Bytes(), nil
+	if w.Err != nil {
+		return nil, w.Err
+	}
+	return w.Bytes(), nil
 }
 
 // decodeList decodes blob into the list of bytes from list bucket index.
 func decodeList(data []byte) (lst [][]byte, err error) {
 	if len(data) == 0 {
 		return nil, nil
 	}
-
-	decoder := gob.NewDecoder(bytes.NewReader(data))
-	if err := decoder.Decode(&lst); err != nil {
-		return nil, err
+	r := io.NewBinReaderFromBuf(data)
+	l := r.ReadVarUint()
+	lst = make([][]byte, l)
+	for i := range lst {
+		lst[i] = r.ReadVarBytes()
+	}
+	if r.Err != nil {
+		return nil, r.Err
 	}
-
 	return lst, nil
 }
 
 // updateBlobovniczaID for existing objects if they were moved from from
 // one blobovnicza to another.
 func updateBlobovniczaID(tx *bbolt.Tx, addr *objectSDK.Address, id *blobovnicza.ID) error {
-	bkt := tx.Bucket(smallBucketName(addr.ContainerID()))
-	if bkt == nil {
-		// if object exists, don't have blobovniczaID and we want to update it
-		// then ignore, this should never happen
-		return ErrIncorrectBlobovniczaUpdate
-	}
-
-	objectKey := objectKey(addr.ObjectID())
-
-	if len(bkt.Get(objectKey)) == 0 {
-		return ErrIncorrectBlobovniczaUpdate
```

Comment:
Without this check `ErrIncorrectBlobovniczaUpdate` is no longer used, so remove it.